### PR TITLE
Db/co2 scd30

### PR DESCRIPTION
**Pull request by danielbernalb**
Added support for CO2 sensor:

Sensirion SCD30 using i2c.

Not readed in the main.cpp and is like an i2c humidity temp sensor

There is a problem with the CO2 reads at main.cpp, in Visual Code everything works ok, compile and works ok on the esp32 but in Platformio online shows an error:
getCO2() and the others 2 routines are not defined in sensors, i prefer to disabled:
// Serial.print (" CO2: " + sensors.getStringCO2());
// Serial.print (" CO2humi: " + String(sensors.getCO2humi()));
// Serial.print (" CO2temp: " + String(sensors.getCO2temp()));

### CODE REVIEW

File: src/Sensors.cpp [hunk 1]
```diff
@@ -19,14 +19,20 @@ void Sensors::loop() {
             if(_onDataCb) _onDataCb();
             dataReady = true;            // only if the main sensor is ready
         }else{
-            if(_onErrorCb)_onErrorCb("-->[W][SENSORS] PM sensor not configured!");
```
Comment (hpsaturn) [inline src/Sensors.cpp:22]: why you remove this line? or why put in the next if?

File: src/Sensors.cpp [hunk 2]
```diff
@@ -19,14 +19,20 @@ void Sensors::loop() {
...
-            dataReady = false;
+                if (isnan(scd30.getCO2())) {
```
Comment (hpsaturn) [inline src/Sensors.cpp:22]: The loop() method is the main sensor method for read all sensors, you don't validate it here, maybe this validation only should happen in the own read method.

File: src/Sensors.cpp [hunk 3]
```diff
@@ -593,6 +654,21 @@ void Sensors::sht31Init() {
...
+  if (scd30.begin() == true) {
+//OPTIONAL configuration
+  //Choose Measurement Interval 
+   //scd30.setMeasurementInterval(2); //Change number of seconds between measurements: 2 to 1800 (30 minutes)
+  //Sealevel
+   //scd30.setAltitudeCompensation(2600); //Set altitude of the sensor in m
+  //Pressure in Boulder, CO is 24.65inHg or 834.74mBar
+   //scd30.setAmbientPressure(1012); //Current ambient pressure in mBar: 700 to 1200
+  //Temperature offset
+   //scd30.setTemperatureOffset(5); //Optionally we can set temperature offset to 5°C
```
Comment (hpsaturn) [inline src/Sensors.cpp:668]: if this comments aren't used, please remove it. I could exposed the sdc30 object public, for example the users could using it.

File: src/Sensors.hpp [hunk 4]
```diff
@@ -139,13 +144,14 @@ class Sensors {
 
     float humi = 0.0;  // % Relative humidity
     float temp = 0.0;  // Temperature (°C)
-    float humi1 = 0.0;  // % Relative humidity
-    float temp1 = 0.0;  // Temperature (°C)
+    float humi1 = 0.0;  // tempo var
```
Comment (hpsaturn) [inline src/Sensors.hpp:147]: the comment is wrong

**Review by hpsaturn — CHANGES_REQUESTED** [review]

File: src/Sensors.cpp (hunk 1 above)
Comment (danielbernalb) [reply src/Sensors.cpp:22]: undo to the original

File: src/Sensors.cpp (hunk 2 above)
Comment (danielbernalb) [reply src/Sensors.cpp:22]: undo to the original

File: src/Sensors.cpp (hunk 3 above)
Comment (danielbernalb) [reply src/Sensors.cpp:668]: is the option for configurate the SCD30, is used few times.
Is optional that is the reason for comments

File: src/Sensors.hpp (hunk 4 above)
Comment (danielbernalb) [reply src/Sensors.hpp:147]: done

File: examples/co2_cm1106/.gitignore [hunk 5]
```diff
@@ -0,0 +1,62 @@
+# Prerequisites
```
Comment (hpsaturn) [inline examples/co2_cm1106/.gitignore:1]: please remove this file

File: examples/co2_cm1106/README.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
+[![PlatformIO](https://github.com/kike-canaries/canairio_sensorlib/workflows/PlatformIO/badge.svg)](https://github.com/kike-canaries/canairio_sensorlib/actions/)
```
Comment (hpsaturn) [inline examples/co2_cm1106/README.md:1]: please remove this file

**Review by hpsaturn — CHANGES_REQUESTED** [review]

File: src/Sensors.cpp (hunk 3 above)
Comment (hpsaturn) [reply src/Sensors.cpp:663]: If we will using it in the future, we should implement it or pass the complete object sdc30 like public, but in master code we don't need this comments for now.

File: examples/co2_scd30/.gitignore [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Prerequisites
```
Comment (hpsaturn) [inline examples/co2_scd30/.gitignore]: please remove this file

File: examples/co2_scd30/README.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
+[![PlatformIO](https://github.com/kike-canaries/canairio_sensorlib/workflows/PlatformIO/badge.svg)](https://github.com/kike-canaries/canairio_sensorlib/actions/)
```
Comment (hpsaturn) [inline examples/co2_scd30/README.md]: please remove this file

**Review by hpsaturn — CHANGES_REQUESTED** [review]

File: examples/co2_cm1106/.gitignore (hunk 5 above)
Comment (danielbernalb) [reply examples/co2_cm1106/.gitignore]: done

File: examples/co2_cm1106/README.md (hunk 6 above)
Comment (danielbernalb) [reply examples/co2_cm1106/README.md]: done

File: examples/co2_scd30/.gitignore (hunk 7 above)
Comment (danielbernalb) [reply examples/co2_scd30/.gitignore]: done

File: examples/co2_scd30/README.md (hunk 8 above)
Comment (danielbernalb) [reply examples/co2_scd30/README.md]: done

File: src/Sensors.cpp (hunk 3 above)
Comment (danielbernalb) [reply src/Sensors.cpp:663]: done

**Review by hpsaturn — APPROVED** [review]

**Pull request merged**